### ISSUE TITLE
Clarify Bower and Spago's current and future support

### ISSUE THREAD

**Issue by actioninja** [JordanMartinez/purescript-jordans-reference#480]
## Context
```
Section:
  Build Tools

Issue Type:
  Bug

Folder/File(s):
  - Entire folder

```

## Body

This whole section paints the distinction between `bower` and `spago` as more of a preferential difference or tools targeted for different purposes. Modern info I am reading on the subject, as well as the total freeze of the `bower` package repo implies that `spago` is the intended primary tool for future usage. Some clarification or deeper research on this subject would be useful.

**Comment by JordanMartinez** [comment]
## More Context on This Situation

I'm sorry that this repo is presently not up-to-date in regards to build tools. (I saw your post in the FP Slack channel). My desire to always be up-to-date sort of backfired here.

For context, this repo was up-to-date regarding the build tools until about February this year. Two PRs were submitted to the PureScript language repo. One introduced the polykinds feature whereas the other introduced the Coercible feature. Both were merged. I expected a `v0.14.0` breaking changes release to be made for the PureScript language in the few weeks that followed, so I updated this repo's syntax in the `v0.14.0-syntax` branch. 

Unfortunately, the release has been significantly delayed for two reasons. First, bugs were found in the Coercible feature. Second, the Bower registry began to no longer accept uploads around February/March as well. So core contributors time has been spent working on both problems.

Thus, I have chosen not to continue updating this repo until after the `v0.14.0` release comes out. Besides the syntax changes, I started fixing a number of other issues in that same PR since I reasons, "At some point, the `v0.14.0` release will come out, which is a breaking change. So I might as well make other major breaking changes in this repo."

## Addressing Some of Your (Likely) Questions/Concerns

So, rather than writing out a new section in #451 where it won't be seen, I'll write something here to help clarify things. First, a few links:

1. See [the original Discourse thread announcing this issue](https://discourse.purescript.org/t/the-bower-registry-is-no-longer-accepting-package-submissions/1103)
2. See [Thomas' "Recommended Tooling" Discourse thread](https://discourse.purescript.org/t/recommended-tooling-for-purescript-applications-in-2019/948)
3. See [paluh's summary for how to publish a new library](https://discourse.purescript.org/t/the-bower-registry-is-no-longer-accepting-package-submissions/1103/16)
4. A [PureScript `master`-branch-compatible package set](https://discourse.purescript.org/t/updating-the-ecosystem-for-upcoming-0-14-0-release/1144/), though I think another one built on top of my last one and fixed a few new issues. Unfortunately, I can't recall where the link is
5. See [Starsuit](https://spacchetti.github.io/starsuit/) for Pursuit-like docs of all packages in the latest (or near-latest) package set.

Second, here's my understanding of where things lie. `spago` should be used as the build tool and dependency manager for the time being. Since `pulp` relied upon `bower`, I don't think either of them will see much use in the future. Core libraries (e.g. those in the `purescript` repo) still use `pulp` + `bower` because the Bower registry still accepts uploads for newer versions of a given library that is already on the registry. In other words, if you try to add a new library to the Bower registry, it will fail. But, if you previously added a library to the registry before this event occurred and wish to push a newer version, it will still succeed. However, I imagine that even newer versions will be prevented at some point.

Currently, I don't believe it's possible to push new libraries' docs to `Pursuit` because `pulp` will verify that they exist on the Bower registry. Since they can't be uploaded to the registry, it will fail. However, the `Starsuit` link I mentioned above provides a way to workaround this problem for the time being. You can also run `spago docs` to get what you need locally.

Lastly, the work on the registry is going slow as they want to make it a great registry. I'm not sure when that will be resolved. Core contributors' time is being split across multiple things during this time period.

**Comment by hdgarrood** [comment]
> Currently, I don't believe it's possible to push new libraries' docs to Pursuit because pulp will verify that they exist on the Bower registry. 

This isn’t the case any more, pulp now only verifies that libraries are in the purescript/registry GitHub repo during `pulp publish`. It doesn’t query the bower registry at all.

**Comment by JordanMartinez** [comment]
Ah, thanks for the update @hdgarrood! I'll push some of my libs' docs then.

**Comment by JordanMartinez** [comment]
This was closed in #546. In sum, while the Bower registry is no longer accepting new uploads, `bower` still works as a dependency manager.
